src/shogun/machine/StochasticGBMachine.cpp

```diff
@@ -273,8 +271,14 @@ CRegressionLabels* CStochasticGBMachine::compute_pseudo_residuals(CRegressionLab
 	return new CRegressionLabels(residuals);
 }
 
-void CStochasticGBMachine::apply_subset(CDenseFeatures<float64_t>* f, CLabels* interf)
+std::tuple<Some<CDenseFeatures<float64_t>>, Some<CRegressionLabels>,
```

Comment:
can you explain the reasonaing here?

Reply:
since view create a new instance, using smart pointers prevent ref/unref

Reply:
doesnt this create the same problem with Some?

Reply:
actually no, this function is only used internally, so we know the static type of return value based on types of args